typing/typecore.ml

```diff
@@ -1958,11 +1958,15 @@ and type_pat_aux
       assert construction_not_used_in_counterexamples;
       let path, new_env =
         !type_open Asttypes.Fresh !env sp.ppat_loc lid in
-      let new_env = ref new_env in
-      type_pat category ~env:new_env p expected_ty ( fun p ->
-        env := Env.copy_local !env ~from:!new_env;
```

Comment:
Since this was the only use of `copy_local`, I think we should remove it from `Env`.